File: playwright/pages/AppLayout.js
import { BasePage } from "./BasePage.js";

export class AppLayout extends BasePage {
  async goToDashboard() {
    await this.testId("nav-dashboard").click();
    await this.page.waitForURL("/");
  }

  async goToUsers() {
    await this.testId("nav-users").click();
    await this.page.waitForURL("**/users");
  }

  async goToProducts() {
    await this.testId("nav-products").click();
    await this.page.waitForURL("**/products");
  }

  async logout() {
    if (await this.isVisible("logout")) {
      await this.testId("logout").click();
    } else {
      await this.testId("logout-mobile").click();
    }
    await this.page.waitForURL("**/login");
  }

  async isNavUsersActive() {
    const className = await this.testId("nav-users").getAttribute("class");
    return className?.includes("bg-indigo-50") ?? false;
  }
}

File: src/test/java/com/automation/um/pages/AppLayout.java
package com.automation.um.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class AppLayout extends BasePage {
  public AppLayout(WebDriver driver) {
    super(driver);
  }

  public void goToDashboard() {
    el("nav-dashboard").click();
    wait.until(ExpectedConditions.urlToBe(com.automation.um.config.Config.BASE_URL + "/"));
  }

  public void goToUsers() {
    el("nav-users").click();
    wait.until(ExpectedConditions.urlContains("/users"));
  }

  public void goToProducts() {
    el("nav-products").click();
    wait.until(ExpectedConditions.urlContains("/products"));
  }

  public void logout() {
    if (isVisible("logout")) {
      el("logout").click();
    } else {
      el("logout-mobile").click();
    }
    wait.until(ExpectedConditions.urlContains("/login"));
  }

  public boolean isNavUsersActive() {
    return el("nav-users").getAttribute("class").contains("bg-indigo-50");
  }
}
